docs/deployment.md

```diff
@@ -78,6 +78,43 @@ Make sure your `package.json` has the `"build"` and `"start"` scripts:
 
 `next build` builds the production application in the `.next` folder. After building, `next start` starts a Node.js server that supports [hybrid pages](/docs/basic-features/pages.md), serving both statically generated and server-side rendered pages.
 
+### Docker Image
+
+Next.js can be deployed to any hosting provider that supports Docker containers. This is the approach you should use when you're deploying to container orchestrators such as Kubernetes or Hashicorp Nomad, or just want to run a docker container in a single node in any cloud provider.
+
+Here is a multi-stage `Dockerfile` using `node:alpine` that you can use:
+
+```Dockerfile
+FROM node:alpine AS builder
+RUN apk add --no-cache libc6-compat
+WORKDIR /app
+COPY package.json yarn.lock ./
+RUN yarn
+COPY . .
+RUN yarn build
```

Comment:
Could you move the building of the package to a separate Docker stage

Related to: https://github.com/vercel/next.js/discussions/16995#discussioncomment-132339

Reply:
Thank you for chiming in here!

Reply:
done! =)